src/rgw/rgw_rest_s3.cc

```diff
@@ -3710,6 +3710,10 @@ void RGWRestoreObj_ObjStore_S3::send_response()
 int RGWDeleteObj_ObjStore_S3::get_params(optional_yield y)
 {
   const char *if_unmod = s->info.env->get("HTTP_X_AMZ_DELETE_IF_UNMODIFIED_SINCE");
+  const char *if_last_mod_time_match = s->info.env->get("HTTP_X_AMZ_IF_MATCH_LAST_MODIFIED_TIME");
+  const char *if_size_match = s->info.env->get("HTTP_X_AMZ_IF_MATCH_SIZE");
+  if_match = s->info.env->get("HTTP_IF_MATCH");
+  if_nomatch = s->info.env->get("HTTP_IF_NONE_MATCH");
```

Comment:
regarding the `if-none-match` header, that isn't mentioned in https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html and boto3 doesn't support the corresponding argument to `delete_object()` so i wasnt able to write tests for it

> client.delete_object(Bucket=bucket, Key=key, IfNoneMatch='*')

fails with

> botocore.exceptions.ParamValidationError: Parameter validation failed:
Unknown parameter in input: "IfNoneMatch", must be one of: Bucket, Key, MFA, VersionId, RequestPayer, BypassGovernanceRetention, ExpectedBucketOwner, IfMatch, IfMatchLastModifiedTime, IfMatchSize